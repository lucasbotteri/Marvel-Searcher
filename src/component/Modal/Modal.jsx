import React from "react";
import styled from "styled-components";
import { faTimes as CloseSvg } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import PropTypes from "prop-types";

const ModalBackground = styled.div`
  background-color: ${(props) => props.theme.opaqueColor};
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
`;

const ModalWrapper = styled.aside`
  background-color: ${(props) => props.theme.primaryColor};
  border-radius: ${(props) => props.theme.radius};
  border: 1px solid ${(props) => props.theme.secondaryColor};
  box-sizing: content-box;
  display: flex;
  flex-direction: column;
  font-size: 19px;
  height: 70%;
  margin: 10% auto;
  padding: 10px;
  overflow: auto;
  width: 450px;

  @media (max-width: 490px) {
    height: 90%;
    margin-top: 10%;
    margin: 0;
    width: 95%;
  }
`;

const ModalContent = styled.div`
  align-self: center;
  padding: 10px;
`;

const CloseIcon = styled(FontAwesomeIcon)`
  align-self: flex-end;
  color: ${(props) => props.theme.secondaryColor};
  cursor: pointer;
  &:hover {
    color: ${(props) => props.theme.opaqueColor};
  }
`;

const ModalTitle = styled.h3`
  align-self: flex-start;
  color: ${(props) => props.theme.primaryFontColor};
  font-family: ${(props) => props.theme.modalFont};
  font-weight: bold;
`;

const Modal = ({ children, title, isShowing, onClose }) => {
  return isShowing ? (
    <ModalBackground>
      <ModalWrapper role="dialog" aria-labelledby="modal-title">
        <CloseIcon icon={CloseSvg} onClick={onClose} aria-label="close modal" />
        <ModalTitle id="modal-title"> {title}</ModalTitle>
        <ModalContent>{children}</ModalContent>
      </ModalWrapper>
    </ModalBackground>
  ) : null;
};

Modal.propTypes = {
  children: PropTypes.element.isRequired,
  title: PropTypes.string.isRequired,
  isShowing: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default Modal;
