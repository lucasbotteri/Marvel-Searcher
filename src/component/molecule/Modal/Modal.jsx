import React from "react";
import styled from "styled-components";
import { faTimes as CloseSvg } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import PropTypes from "prop-types";

export const ModalBackground = styled.div`
  background-color: ${(props) => props.theme.opaqueColor};
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
`;

export const ModalWrapper = styled.aside`
  background-color: ${(props) => props.theme.primaryColor};
  border-radius: ${(props) => props.theme.radius};
  border: 1px solid ${(props) => props.theme.secondaryColor};
  box-sizing: content-box;
  display: flex;
  flex-direction: column;
  font-size: 19px;
  height: 70%;
  margin: 10% auto;
  padding: 10px;
  width: 400px;

  @media (max-width: 425px) {
    height: 100%;
    margin: 45px 0;
    width: 95%;
  }
`;

export const ModalContent = styled.div`
  align-self: center;
  padding: 10px;
  overflow: auto;
`;

export const CloseIcon = styled(FontAwesomeIcon)`
  align-self: flex-end;
  color: ${(props) => props.theme.secondaryColor};
  cursor: pointer;
  &:hover {
    color: ${(props) => props.theme.opaqueColor};
  }
`;

export const ModalTitle = styled.h3`
  align-self: flex-start;
  color: ${(props) => props.theme.primaryFontColor};
  font-family: ${(props) => props.theme.modalFont};
  font-weight: bold;
`;

const Modal = ({ children, title, onClose }) => {
  return (
    <ModalBackground>
      <ModalWrapper role="dialog" aria-labelledby="modal-title">
        <CloseIcon icon={CloseSvg} onClick={onClose} aria-label="close modal" />
        <ModalTitle id="modal-title"> {title}</ModalTitle>
        <ModalContent>{children}</ModalContent>
      </ModalWrapper>
    </ModalBackground>
  );
};

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default Modal;
