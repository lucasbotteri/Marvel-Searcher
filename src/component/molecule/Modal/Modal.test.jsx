import React from "react";
import { shallow } from "enzyme";
import Modal, { CloseIcon } from "./Modal";

describe("ComicDetail", () => {
  it("Should render without crashing", () => {
    const wrapper = shallow(
      <Modal title="title" onClose={() => {}}>
        <div>Test</div>
      </Modal>
    );
    expect(wrapper).toBeTruthy();
  });

  it("Should render content", () => {
    const wrapper = shallow(
      <Modal title="title" onClose={() => {}}>
        <div>Test</div>
      </Modal>
    );
    expect(wrapper.containsMatchingElement(<div>Test</div>)).toBeTruthy();
  });

  it("Should render title", () => {
    const wrapper = shallow(
      <Modal title="title" onClose={() => {}}>
        <div>Test</div>
      </Modal>
    );
    expect(wrapper.containsMatchingElement("title")).toBeTruthy();
  });

  it("Should call onClose", () => {
    const onClose = jest.fn();
    const wrapper = shallow(
      <Modal title="title" onClose={onClose}>
        <div>Test</div>
      </Modal>
    );
    const closeIcon = wrapper.find(CloseIcon);
    closeIcon.simulate("click");
    expect(onClose).toHaveBeenCalled();
  });

  it("Should match snapshot", () => {
    const wrapper = shallow(
      <Modal title="title" onClose={() => {}}>
        <div>Test</div>
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
