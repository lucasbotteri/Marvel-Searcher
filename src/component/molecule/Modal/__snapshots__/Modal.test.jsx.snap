// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicDetail Should match snapshot 1`] = `ShallowWrapper {}`;
